client/components/seo/meta-title-editor/index.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { Component, PropTypes } from 'react';
+import get from 'lodash/get';
+import identity from 'lodash/identity';
+import isString from 'lodash/isString';
+import isUndefined from 'lodash/isUndefined';
+import map from 'lodash/map';
+import { connect } from 'react-redux';
+
+import SegmentedControl from 'components/segmented-control';
+import TokenField from 'components/token-field';
+import { localize } from 'i18n-calypso';
+
+import {
+	nativeToRaw,
+	rawToNative,
+	removeBlanks
+} from './mappings';
+
+const titleTypes = translate => [
+	{ value: 'frontPage', label: translate( 'Front Page' ) },
+	{ value: 'posts', label: translate( 'Posts' ) },
+	{ value: 'pages', label: translate( 'Pages' ) },
+	{ value: 'groups', label: translate( 'Categories & Tags' ) },
+	{ value: 'archives', label: translate( 'Archives' ) }
+];
+
+const getValidTokens = translate => ( {
+	siteName: translate( 'Site Name' ),
+	tagline: translate( 'Tagline' ),
+	postTitle: translate( 'Post Title' ),
+	pageTitle: translate( 'Page Title' ),
+	groupTitle: translate( 'Category/Tag Title' ),
+	date: translate( 'Date' )
+} );
+
+const tokenMap = {
+	frontPage: [ 'siteName', 'tagline' ],
+	posts: [ 'siteName', 'tagline', 'postTitle' ],
+	pages: [ 'siteName', 'tagline', 'pageTitle' ],
+	groups: [ 'siteName', 'tagline', 'groupTitle' ],
+	archives: [ 'siteName', 'tagline', 'date' ]
+};
+
+const tokenize = translate => s => {
```

Comment:
Well if it makes you feel better it is part of our [naming conventions](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#naming-conventions)  ;) And on my first pass, I did not realize what k and s stood for since I  was admiring your ascii art. 

I'm fine with a single letter variable name if the language has some obvious naming conventions for it, or it's math, ie: `transform( a, b, c, d, e, f )`